### ISSUE TITLE
Excessive dependencies

### ISSUE THREAD

**Issue by ben-manes** [intuit/QuickBooks-V3-Java-SDK#23]
The `ipp-v3-java-devkit-pom` includes many compile time dependencies, such as the `maven-shade-plugin`, which make no sense. I am debugging an ASM conflict failure and this library is one of them using a very old version without shading, despite not seeing that in its usage. Please declare your dependencies correctly.

```
+--- com.intuit.quickbooks-online:ipp-v3-java-data:2.9.5
|    +--- joda-time:joda-time:2.1 -> 2.9.7
|    +--- org.jvnet.jaxb2_commons:jaxb2-commons-lang:2.4
|    |    \--- org.apache.commons:commons-lang3:3.2.1 -> 3.6
|    +--- javax.xml.bind:jaxb-api:2.2.12
|    +--- com.fasterxml.jackson.core:jackson-core:2.8.8
|    +--- com.fasterxml.jackson.core:jackson-databind:2.8.8 (*)
|    +--- org.jvnet.jaxb2_commons:jaxb2-basics-runtime:1.11.1
|    +--- org.slf4j:slf4j-api:1.6.4 -> 1.7.25
|    +--- org.apache.httpcomponents:httpclient:4.5.2 (*)
|    +--- org.apache.httpcomponents:httpcore:4.4.5
|    +--- oauth.signpost:signpost-core:1.2.1.1
|    |    \--- commons-codec:commons-codec:1.3 -> 1.10
|    +--- oauth.signpost:signpost-commonshttp4:1.2
|    |    +--- oauth.signpost:signpost-core:1.2 -> 1.2.1.1 (*)
|    |    +--- org.apache.httpcomponents:httpcore:4.0.1 -> 4.4.5
|    |    \--- org.apache.httpcomponents:httpclient:4.0.1 -> 4.5.2 (*)
|    +--- commons-logging:commons-logging:1.1.1 -> 1.2
|    +--- commons-configuration:commons-configuration:1.6
|    |    +--- commons-collections:commons-collections:3.2.1 -> 3.2.2
|    |    +--- commons-lang:commons-lang:2.4
|    |    +--- commons-logging:commons-logging:1.1.1 -> 1.2
|    |    +--- commons-digester:commons-digester:1.8 -> 1.8.1 (*)
|    |    \--- commons-beanutils:commons-beanutils-core:1.8.0
|    |         \--- commons-logging:commons-logging:1.1.1 -> 1.2
|    +--- com.google.code.gson:gson:2.8.1
|    +--- org.apache.maven.plugins:maven-shade-plugin:1.7
|    |    +--- org.apache.maven:maven-plugin-api:2.0.6
|    |    +--- org.apache.maven:maven-model:2.0.6 -> 2.0.8
|    |    |    \--- org.codehaus.plexus:plexus-utils:1.4.6 -> 1.5.5
|    |    +--- org.apache.maven:maven-project:2.0.6 -> 2.0.8
|    |    |    +--- org.apache.maven:maven-settings:2.0.8
|    |    |    |    +--- org.apache.maven:maven-model:2.0.8 (*)
|    |    |    |    +--- org.codehaus.plexus:plexus-utils:1.4.6 -> 1.5.5
|    |    |    |    \--- org.codehaus.plexus:plexus-container-default:1.0-alpha-9-stable-1
|    |    |    |         +--- junit:junit:3.8.1
|    |    |    |         +--- org.codehaus.plexus:plexus-utils:1.0.4 -> 1.5.5
|    |    |    |         \--- classworlds:classworlds:1.1-alpha-2
|    |    |    +--- org.apache.maven:maven-profile:2.0.8
|    |    |    |    +--- org.apache.maven:maven-model:2.0.8 (*)
|    |    |    |    +--- org.codehaus.plexus:plexus-utils:1.4.6 -> 1.5.5
|    |    |    |    \--- org.codehaus.plexus:plexus-container-default:1.0-alpha-9-stable-1 (*)
|    |    |    +--- org.apache.maven:maven-model:2.0.8 (*)
|    |    |    +--- org.apache.maven:maven-artifact-manager:2.0.8
|    |    |    |    +--- org.apache.maven:maven-repository-metadata:2.0.8
|    |    |    |    |    \--- org.codehaus.plexus:plexus-utils:1.4.6 -> 1.5.5
|    |    |    |    +--- org.codehaus.plexus:plexus-utils:1.4.6 -> 1.5.5
|    |    |    |    +--- org.apache.maven:maven-artifact:2.0.8
|    |    |    |    |    \--- org.codehaus.plexus:plexus-utils:1.4.6 -> 1.5.5
|    |    |    |    +--- org.codehaus.plexus:plexus-container-default:1.0-alpha-9-stable-1 (*)
|    |    |    |    \--- org.apache.maven.wagon:wagon-provider-api:1.0-beta-2
|    |    |    |         \--- org.codehaus.plexus:plexus-utils:1.0.4 -> 1.5.5
|    |    |    +--- org.apache.maven:maven-plugin-registry:2.0.8
|    |    |    |    +--- org.codehaus.plexus:plexus-utils:1.4.6 -> 1.5.5
|    |    |    |    \--- org.codehaus.plexus:plexus-container-default:1.0-alpha-9-stable-1 (*)
|    |    |    +--- org.codehaus.plexus:plexus-utils:1.4.6 -> 1.5.5
|    |    |    +--- org.apache.maven:maven-artifact:2.0.8 (*)
|    |    |    \--- org.codehaus.plexus:plexus-container-default:1.0-alpha-9-stable-1 (*)
|    |    +--- org.apache.maven:maven-artifact:2.0.6 -> 2.0.8 (*)
|    |    +--- org.codehaus.plexus:plexus-container-default:1.0-alpha-9-stable-1 (*)
|    |    +--- org.codehaus.plexus:plexus-utils:1.5.5
|    |    +--- asm:asm:3.3.1
|    |    +--- asm:asm-commons:3.3.1
|    |    |    \--- asm:asm-tree:3.3.1
|    |    |         \--- asm:asm:3.3.1
|    |    +--- org.jdom:jdom:1.1
|    |    +--- org.apache.maven.shared:maven-dependency-tree:1.1
|    |    |    \--- org.apache.maven:maven-project:2.0.8 (*)
|    |    +--- org.vafer:jdependency:0.7
|    |    |    +--- commons-io:commons-io:1.3.2 -> 2.5
|    |    |    +--- asm:asm:3.2 -> 3.3.1
|    |    |    +--- asm:asm-analysis:3.2
|    |    |    |    \--- asm:asm-tree:3.2 -> 3.3.1 (*)
|    |    |    +--- asm:asm-commons:3.2 -> 3.3.1 (*)
|    |    |    +--- asm:asm-util:3.2
|    |    |    |    \--- asm:asm-tree:3.2 -> 3.3.1 (*)
|    |    |    \--- asm:asm-tree:3.2 -> 3.3.1 (*)
|    |    \--- org.codehaus.plexus:plexus-component-annotations:1.5.5
|    +--- cglib:cglib:2.2.2
|    |    \--- asm:asm:3.3.1
|    +--- asm:asm-commons:3.3.1 (*)
|    +--- net.sf.kxml:kxml2:2.2.2
|    |    \--- xmlpull:xmlpull:1.1.3.1
|    +--- javax.mail:mailapi:1.4.3 (*)
|    +--- com.sun.xml.bind:jaxb-impl:2.2.11
|    +--- com.sun.xml.bind:jaxb-core:2.2.11
|    +--- com.fasterxml.jackson.core:jackson-annotations:2.8.8
|    +--- com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.8.8 (*)
|    +--- com.fasterxml.jackson.module:jackson-module-jaxb-annotations:2.8.8 (*)
|    \--- com.fasterxml.jackson.datatype:jackson-datatype-joda:2.8.8 (*)
```

**Comment by ben-manes** [comment]
Also please upgrade cglib to use a modern version of ASM. This is again causing frustrations due to ASM conflicts. You can use Maven's version plugin to ensure that your dependencies are up to date. Alternatively do not use low-level, error prone libraries if you do not plan on maintaining compatibility.

**Comment by diana-derose** [comment]
Hi Ben - Thanks for the feedback. We are aware that the SDK currently uses older versions of many libraries some of which are probably not needed and we are working on updating them one at a time. We are already working on removal of "maven-shade-plugin" for the next release and asm/cglib upgrade is in the roadmap. I'll keep you posted.

**Comment by ben-manes** [comment]
Thanks!

**Comment by davidseywald** [comment]
Same request for the java mail api (remove or update if possible). Thanks.

**Comment by diana-derose** [comment]
Hi @davidseywald  - We'll update it to com.sun.mail v1.6.1 in the next release.

**Comment by Bigtuna00** [comment]
Pull request #45 hopefully eliminates the build problems.

**Comment by diana-derose** [comment]
Released this in v4.1.0
